presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2068,18 +2100,38 @@ public void run(ExtendedHiveMetastore metastore)
                     // Not deleting the table may leave garbage behind. The former is much more dangerous than the latter.
                     // Therefore, the table is not considered added.
                 }
+
                 if (!done) {
                     throw e;
                 }
             }
+            tableCreated = true;
```

Comment:
Put this at the end of the try block. It's hard to see since the catch block is huge

Reply:
Putting it at the end of the outer try block will be incorrect as we should also set `tableCreated=true` in some cases on exception. It's also not correct to put it at the end of the inner try block either.